gqlalchemy/graph_algorithms/query_modules.py

```diff
@@ -0,0 +1,153 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Tuple, List
+
+QM_KEY_NAME = "name"
+QM_KEY_VALUE = "value"
+QM_KEY_DEFAULT = "default"
+QM_KEY_TYPE = "type"
+
+QM_FIELD_NAME = "name"
+QM_FIELD_IS_EDITABLE = "is_editable"
+QM_FIELD_IS_WRITE = "is_write"
+QM_FIELD_PATH = "path"
+QM_FIELD_SIGNATURE = "signature"
+
+
+class QueryModule:
+    """Class representing a single MAGE query module."""
+
+    def __init__(self, module_dict: Dict) -> None:
+        arguments, returns = parse_query_module_signature(module_dict[QM_FIELD_SIGNATURE])
+
+        self.name = module_dict[QM_FIELD_NAME]
+        self.is_editable = module_dict[QM_FIELD_IS_EDITABLE]
+        self.is_write = module_dict[QM_FIELD_IS_WRITE]
+        self.path = module_dict[QM_FIELD_PATH]
+        self.signature = module_dict[QM_FIELD_SIGNATURE]
+        self.arguments = arguments
+        self.returns = returns
+
+    def __str__(self) -> str:
+        return self.name
+
+    def set_argument_values(self, **kwargs) -> None:
+        """Set values for QueryModule arguments so the module can be called.
+
+        Kwargs:
+            Named arguments in self.arguments.
+
+        Raises:
+            KeyError: Passed an argument not in the self.arguments list.
+        """
+        for argument_name in kwargs.keys():
+            has_arg = False
+            for argument_dict in self.arguments:
+                if argument_dict[QM_KEY_NAME] == argument_name:
+                    argument_dict[QM_KEY_VALUE] = str(kwargs[argument_name])
+                    has_arg = True
+            if not has_arg:
+                raise KeyError(f"{argument_name} is not an argument in this query module.")
+
+    def get_arguments_for_call(self) -> str:
+        """return inputs in form "value1, value2, ..." for QueryBuilder call()
+        method.
+
+        Raises:
+            KeyError: Cannot get all values of arguments because one or more is
+            not set.
+        """
+        arguments_str = ""
+        for argument_dict in self.arguments:
+            if QM_KEY_VALUE in argument_dict:
+                val = argument_dict[QM_KEY_VALUE]
+            elif QM_KEY_DEFAULT in argument_dict:
+                val = argument_dict[QM_KEY_DEFAULT]
+            else:
+                raise KeyError(f"{argument_dict[QM_KEY_NAME]} has no value set.")
+
+            if argument_dict[QM_KEY_TYPE] == "STRING":
+                arguments_str += '"' + val + '"'
+            else:
+                arguments_str += val
+
+            arguments_str += ", "
+
+        arguments_str = arguments_str[:-2]
+
+        return arguments_str
```

Comment:
`return arguments_str[:-2]`

Reply:
Wouldn't agree. Returning plain result that is already processed is better for debugging

Reply:
Additional assignment is not necessary here, but ok